ouroboros-consensus-cardano/ouroboros-consensus-cardano.cabal

```diff
@@ -146,7 +146,10 @@ library
     , sop-core                      ^>=0.5
     , sop-extras                    ^>=0.1
     , strict-sop-core               ^>=0.1
-    , text                          ^>=2.0
+    -- text-2 requires some C++ parts, and won't build in
+    -- some cross situations. So we continue to allow 
+    -- text-1 for now
+    , text                          >= 1 && < 2.2
```

Comment:
I wonder if we could gate this behind a conditional to only enable it in cross builds. I don't know what the `os` flag says when cross compiling.

Reply:
I'm unsure if that buys you anything. In both cases you need to retain compatibility across the text versions, this way is if anything easier to test because you don't have to trigger the cross conditional.

Reply:
It is indeed easier to test, but are we ever going to test on Linux for earlier versions? I don't think I have seen such a case yet

Reply:
Unless something goes wrong on the compilation pipeline but in that case you can do `cabal build --allow-older --constraint "text ==<problematic-version>`

Reply:
Actually the `os` flag says:

> The argument is tested against System.Info.os on the target system. 

So I think `os(windows)` will evaluate to true when cross-compiling for windows.

Reply:
Sure, but what does restricting it get you? The code has to work for the wider text range regardless! If you had a situation where, say, you had some code that was only conditionally included when not building on windows that required a newer text then I would see the point, but here I'm not sure that I do.